pyogrio/geopandas.py

```diff
@@ -97,13 +98,23 @@ def read_dataframe(
         and used by GDAL, only geometries that intersect this bbox will be
         returned; if GEOS is not available or not used by GDAL, all geometries
         with bounding boxes that intersect this bbox will be returned.
+        Cannot be combined with ``mask`` keyword.
+    mask : Shapely geometry, optional (default: None)
+        If present, will be used to filter records whose geometry intersects
+        this geometry.  This must be in the same CRS as the dataset.  If GEOS is
+        present and used by GDAL, only geometries that intersect this geometry
+        will be returned; if GEOS is not available or not used by GDAL, all
+        geometries with bounding boxes that intersect the bounding box of this
+        geometry will be returned.  Requires Shapely >= 2.0.
+        Cannot be combined with ``bbox`` keyword.
     fids : array-like, optional (default: None)
         Array of integer feature id (FID) values to select. Cannot be combined
-        with other keywords to select a subset (``skip_features``, ``max_features``,
-        ``where``, ``bbox`` or ``sql``). Note that the starting index is driver and file
-        specific (e.g. typically 0 for Shapefile and 1 for GeoPackage, but can
-        still depend on the specific file). The performance of reading a large
-        number of features usings FIDs is also driver specific.
+        with other keywords to select a subset (``skip_features``,
+        ``max_features``, ``where``, ``bbox``, ``mask``, or ``sql``). Note that
+        the starting index is driver and file specific (e.g. typically 0 for
+        Shapefile and 1 for GeoPackage, but can still depend on the specific
+        file). The performance of reading a large number of features usings FIDs
+        is also driver specific.
     sql : str, optional (default: None)
         The SQL statement to execute. Look at the sql_dialect parameter for
         more information on the syntax to use for the query. When combined
```

Comment:
Probably ``mask`` will behave the same as ``bbox`` in combination with ``sql``? Best to test it though + add here in doc?

Same for ``where``?

Reply:
I added tests, but I'm not clear on what we should note in the docstring.  Do you mean state that you can combine `bbox` or `mask` with `sql` or `where` parameters?

Reply:
It's apparently not possible to add review comments in code/doc that hasn't been touched... but in the sql documentation `or mask` could  be added to the list of keywords in this listed keywords:
```
... When combined with other keywords like ``columns``, ``skip_features``, ``max_features``, ``where`` or ``bbox``, those are applied after the SQL query. ...
```